src/test/java/teammates/test/cases/storage/InstructorsDbTest.java

```diff
@@ -328,6 +330,7 @@ public void testUpdateInstructorByGoogleId() throws Exception {
         assertEquals("new role", instructorUpdated.role);
         assertFalse(instructorUpdated.isDisplayedToStudents);
         assertEquals("New Displayed Name", instructorUpdated.displayedName);
+        assertTrue(instructorUpdated.hasObserverPrivileges());
```

Comment:
@leeyimin I tried that out and it doesn't return true for more privileged roles but I guess I'll add this in any way to protect from any future changes.